### PR TITLE
`bevy_color`: Added `Hsva` and `Hwba` Models

### PR DESCRIPTION
**Pull request by bushrat011899**
# Objective

- Improve compatibility with CSS Module 4
- Simplify `Hsla` conversion functions

## Solution

- Added `Hsva` which implements the HSV color model.
- Added `Hwba` which implements the HWB color model.
- Updated `Color` and `LegacyColor` accordingly.

## Migration Guide

- Convert `Hsva` / `Hwba` to either `Hsla` or `Srgba` using the provided `From` implementations and then handle accordingly.

## Notes

While the HSL color space is older than HWB, the formulation for HWB is more directly related to RGB. Likewise, HSV is more closely related to HWB than HSL. This makes the conversion of HSL to/from RGB more naturally represented as the compound operation HSL <-> HSV <-> HWB <-> RGB. All `From` implementations for HSL, HSV, and HWB have been designed to take the shortest path between itself and the target space.

### CODE REVIEW

File: crates/bevy_color/src/lib.rs [hunk 1]
```diff
@@ -31,6 +33,11 @@
 //! A gradient in HSL space from red to violet will produce a rainbow. The LCH color space is
 //! more perceptually accurate than HSL, but is less intuitive to work with.
 //!
+//! HSV and HWB are very closely related to HSL in their derivation, having identical definitions for
+//! hue. Where HSL uses saturation and lightness, HSV uses a slightly modified definition of saturation,
+//! and an analog of lightness in the form of value. Contrastingly, HWB instead uses whiteness and blackness
```
Comment (alice-i-cecile) [inline crates/bevy_color/src/lib.rs:38]:
```suggestion
//! and an analog of lightness in the form of value. In contrast, HWB instead uses whiteness and blackness
```

File: crates/bevy_color/src/hsva.rs [hunk 2]
```diff
@@ -0,0 +1,213 @@
+use crate::{Alpha, Hwba, Lcha, LinearRgba, Oklaba, Srgba, StandardColor, Xyza};
+use bevy_reflect::{Reflect, ReflectDeserialize, ReflectSerialize};
+use serde::{Deserialize, Serialize};
+
+/// Color in Hue-Whiteness-Blackness color space with alpha
```
Comment (alice-i-cecile) [inline crates/bevy_color/src/hsva.rs:5]: I'd like to see a link to further reading on both of these color models.

File: crates/bevy_color/src/hsla.rs [hunk 3]
```diff
@@ -127,91 +127,85 @@ impl Luminance for Hsla {
...
+        Hsva {
             hue,
             saturation,
-            lightness,
+            value,
             alpha,
-        }: Hsla,
+        }: Hsva,
     ) -> Self {
-        // https://en.wikipedia.org/wiki/HSL_and_HSV#HSL_to_RGB
```
Comment (alice-i-cecile) [inline crates/bevy_color/src/hsla.rs:173]: We should avoid losing this link (or replace it with another derivation if needed).

Comment (bushrat011899) [reply crates/bevy_color/src/hsla.rs:173]: Good point! I'll update all 6 core `From` implementations across HSL/HSV/HWB to explicitly include links to where these formulae come from.

File: crates/bevy_color/src/hsva.rs (hunk 2 above)
Comment (bushrat011899) [reply crates/bevy_color/src/hsva.rs:5]: For all three of HSL/HSV/HWB I'll include links to their respective Wikipedia articles. For HWB I've also included a link to the original defining paper via archive.org, which is what I've based my conversion implementations on.

File: crates/bevy_color/src/hsla.rs (hunk 3 above)
Comment (alice-i-cecile) [reply crates/bevy_color/src/hsla.rs:173]: You're the best :)

**Review by alice-i-cecile — APPROVED** [review]

**Review by afonsolage — APPROVED** [review]

**Pull request merged**